### PR TITLE
Add pipeline for kyma cli alpha feature - upgrade

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**

Changes proposed in this pull request:

- Add new pipeline for cli alpha features
- Pipeline workflow:
  - Compile cli from master
  - Provision gardener cluster from cli
  - Install kyma  stable release with `alpha deploy`
  - Run test suite
  - Update kyma to latest master
  - Run test suite

> **NOTE:** console-web tests are failing constantly and were disabled for this pipeline

**Related issue(s)**

- Implements #3057

### CODE REVIEW

**Review by colunira — APPROVED** [review]

File: prow/scripts/cluster-integration/kyma-cli-alpha-upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+export TEST_INFRA_SOURCES_DIR="${KYMA_PROJECT_DIR}/test-infra"
+export TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS="${TEST_INFRA_SOURCES_DIR}/prow/scripts/cluster-integration/helpers"
+
+# shellcheck source=prow/scripts/lib/gardener/gcp.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/gcp.sh"
+# shellcheck disable=SC1090
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/testing-helpers.sh"
+# shellcheck source=prow/scripts/lib/utils.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/utils.sh"
+
```
Comment (Halamix2) [inline prow/scripts/cluster-integration/kyma-cli-alpha-upgrade.sh:37]: Direct source of log library would be better than relying on other libraries to import it
```bash
# shellcheck source=prow/scripts/lib/log.sh
source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/log.sh"
```

File: prow/scripts/lib/cli-alpha.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+# cli-alpha::deploy starts Kyma installation using the alpha deploy command
+#
+# Arguments:
+#	$1 - Path to local resource directory
+#	$2 - Path to local components.yaml file
+function cli-alpha:deploy {
```
Comment (Ressetkk) [inline prow/scripts/lib/cli-alpha.sh:8]: Please use a naming scheme `lib_name::function` and add second `:`

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**